x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/advanced_step/hyper_parameters.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, Fragment } from 'react';
+import { EuiFieldNumber, EuiFlexItem, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { CreateAnalyticsFormProps } from '../../../analytics_management/hooks/use_create_analytics_form';
+
+const MAX_TREES_LIMIT = 2000;
+
+export const HyperParameters: FC<CreateAnalyticsFormProps> = ({ actions, state }) => {
+  const { setFormState } = actions;
+
+  const { featureBagFraction, maxTrees, randomizeSeed } = state.form;
+
+  return (
+    <Fragment>
+      <EuiFlexItem style={{ minWidth: '30%' }}>
+        <EuiFormRow
+          label={i18n.translate('xpack.ml.dataframe.analytics.create.lambdaLabel', {
+            defaultMessage: 'Lambda',
+          })}
+          helpText={i18n.translate('xpack.ml.dataframe.analytics.create.lambdaHelpText', {
+            defaultMessage:
+              'Regularization parameter to prevent overfitting on the training data set. Must be a non negative value',
+          })}
+        >
+          <EuiFieldNumber
+            aria-label={i18n.translate('xpack.ml.dataframe.analytics.create.lambdaInputAriaLabel', {
+              defaultMessage:
+                'Regularization parameter to prevent overfitting on the training data set.',
+            })}
+            data-test-subj="mlAnalyticsCreateJobFlyoutLambdaInput"
+            onChange={(e) => setFormState({ lambda: +e.target.value })}
+            step={0.001}
+            min={0}
+          />
+        </EuiFormRow>
+      </EuiFlexItem>
+      <EuiFlexItem style={{ minWidth: '30%' }}>
+        <EuiFormRow
+          label={i18n.translate('xpack.ml.dataframe.analytics.create.maxTreesLabel', {
+            defaultMessage: 'Max trees',
+          })}
+          helpText={i18n.translate('xpack.ml.dataframe.analytics.create.maxTreesText', {
+            defaultMessage:
+              'The maximum number of trees the forest is allowed to contain. Must be an integer',
+          })}
+        >
+          <EuiFieldNumber
```

Comment:
Good catch @peteharverson 👍 This was happening because the empty string was converted to a number - which gives a 0. Updated to set the value as undefined if field is empty.

Reply:
If I set a value for one of the hyper parameters, e.g. max trees, then hit Continue, then go back by clicking 'Edit', and expand the hyper parameters, the value I put in for max trees is no longer showing up.

Reply:
Fixed in cbdf3d5495501b77667dcb273df7817594782d25